djangoRT/views.py

```diff
@@ -6,200 +6,236 @@
+        files = [(attachment.name, attachment, mime_type)]
+        success = rt.replyToTicket(ticket_id, files=files)
+        if not success:
+            logger.error(f"Error adding attachment to #{ticket_id}")
+
+    messages.success(request, (
+        f"Ticket #{ticket_id} has been successfully created. "
+        "We will respond to your request as soon as possible."))
+
+    return ticket_id
+
+
+def ticketcreate(request):
+    # Don't require login, be nice and take to guest ticket page.
+    if not request.user.is_authenticated():
+        return HttpResponseRedirect(reverse("djangoRT:ticketcreateguest"))
+
+    if request.method == "POST":
+        form = forms.TicketForm(request.POST, request.FILES)
+        ticket_id = _handle_ticket_form(request, form)
+        if ticket_id is not None:
+            return HttpResponseRedirect(reverse("djangoRT:ticketdetail", args=[ticket_id]))
     else:
         form = forms.TicketForm(initial={
-            'email' : request.user.email,
-            'first_name' : request.user.first_name,
-            'last_name' : request.user.last_name
+            "email": request.user.email,
+            "first_name": request.user.first_name,
+            "last_name": request.user.last_name
         })
-    return render(request, 'djangoRT/ticketCreate.html', { 'form' : form })
 
-def ticketcreateguest(request):
-    rt = rtUtil.DjangoRt()
+    return render(request, "djangoRT/ticketCreate.html", {"form": form})
 
-    data = {}
+
+def ticketcreateguest(request):
     if request.user.is_authenticated():
-        return HttpResponseRedirect( reverse( 'djangoRT:ticketcreate'), )
+        return HttpResponseRedirect(reverse("djangoRT:ticketcreate"))
 
-    if request.method == 'POST':
+    if request.method == "POST":
         form = forms.TicketGuestForm(request.POST, request.FILES)
-
-        if form.is_valid():
-            ticket = rtModels.Ticket(subject = form.cleaned_data['subject'],
-                    problem_description = form.cleaned_data['problem_description'],
-                    requestor = form.cleaned_data['email'])
-            ticket_id = rt.createTicket(ticket)
-
-            if ticket_id > -1:
-                if 'attachment' in request.FILES:
-                    rt.replyToTicket(ticket_id, files=([request.FILES['attachment'].name, request.FILES['attachment'], mimetypes.guess_type(request.FILES['attachment'].name)],))
-                messages.add_message(request, messages.SUCCESS, 'Ticket #%s has been successfully created. We will respond to your request as soon as possible.' % ticket_id)
-                form = forms.TicketGuestForm()
-                return render(request, 'djangoRT/ticketCreateGuest.html', { 'form': form })
-            else:
-                # make this cleaner probably
-                messages.error('An unexpected error occurred while creating your ticket. Please try again.')
-                data['first_name'] = form.cleaned_data['first_name']
```

Comment:
This kind of thing is not necessary. The form will default to the values it found on POST and if it's not valid, it will re-render those values back in to the form with error messages. This is what Django is designed to do.